### PR TITLE
hyprland: 0.43.0 -> 0.44.0, hyprpicker: 0.3.0 -> 0.4.1

### PR DESCRIPTION
**Pull request by fufexan**
Update hyprland and hyprpicker. More details in the commit messages.

[Hyprland CHANGELOG](https://github.com/hyprwm/Hyprland/releases/v0.44.0)
[Hyprpicker CHANGELOG](https://github.com/hyprwm/hyprpicker/releases/tag/v0.4.1)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/hy/hyprland/package.nix [hunk 1]
```diff
@@ -45,30 +46,50 @@
...
+    flatten
+    concatLists
+    optionals
+    ;
+  inherit (lib.strings)
+    makeBinPath
+    optionalString
+    mesonBool
+    mesonEnable
+    ;
```
Comment (NotAShelf) [inline pkgs/by-name/hy/hyprland/package.nix:66]: words can hardly express how much I hate this

**Review by NotAShelf — APPROVED** [review]
lgtm, love u fuf

File: pkgs/by-name/hy/hyprland/package.nix (hunk 1 above)
Comment (fufexan) [reply pkgs/by-name/hy/hyprland/package.nix:66]: The art of the glorious nixfmt.

**Comment by Atemu** [comment]
Could you separate out the reformat part into a separate commit? It's impossible to tell what's going on otherwise.

**Comment by JohnRTitor** [comment]
Seperate the mold linker commit as well.

**Comment by Atemu** [comment]
I meant the hyprpicker commit.

It's fine to bundle changes and minor fixups related to an update into the update commit but automated reformats generally make it impossible to tell what other changes have actually taken place, so please separate that out into another commit.

File: pkgs/by-name/hy/hyprland/package.nix [hunk 2]
```diff
@@ -45,30 +46,52 @@
...
+
+  info = fromJSON (readFile ./info.json);
+
+  # possibility to add more adapters in the future, such as keepDebugInfo,
+  # which would be controlled by the `debug` flag
+  adapters = flatten [
+    stdenvAdapters.useMoldLinker
+  ];
```
Comment (Atemu) [inline pkgs/by-name/hy/hyprland/package.nix:74]: Nit: This strikes me as odd: Why do you flatten a list that only contains a function? This is a no-op.

File: pkgs/by-name/hy/hyprland/package.nix [hunk 3]
```diff
@@ -45,30 +46,52 @@
...
+    optionals
+    ;
+  inherit (lib.strings)
+    makeBinPath
+    optionalString
+    mesonBool
+    mesonEnable
+    ;
+
+  info = fromJSON (readFile ./info.json);
```
Comment (Atemu) [inline pkgs/by-name/hy/hyprland/package.nix:68]: Nit: lib.importJSON

**Review by Atemu — APPROVED** [review]
A few nits, otherwise diff LGTM.

File: pkgs/by-name/hy/hyprland/package.nix (hunk 2 above)
Comment (fufexan) [reply pkgs/by-name/hy/hyprland/package.nix:74]: Leftover from a previous attempt at also adding `(optional debug keepDebugInfo)`. I will remove it.

File: pkgs/by-name/hy/hyprland/package.nix (hunk 3 above)
Comment (fufexan) [reply pkgs/by-name/hy/hyprland/package.nix:68]: TIL

**Comment by Atemu** [comment]
You forgot to update the info.json.

This would have been caught by a simple `Hyprland --version` check, so perhaps this would be a good time as any to introduce it.

**Comment by Atemu** [comment]
https://github.com/NixOS/nixpkgs/pull/347624

**Comment by fufexan** [comment]
> You forgot to update the info.json.

The problem is I forgot to update through the update script, and manually replaced the version and hash instead. Muscle memory.

**Pull request merged**